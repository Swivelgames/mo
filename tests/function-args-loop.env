arr=(
    "test1"
    "test2"
)

testArgs() {
    value=$(cat)
    echo "$value - $MO_FUNCTION_ARGS"
}
