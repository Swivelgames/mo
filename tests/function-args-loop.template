{{#arr1}}
{{#arr2}}
No args: {{$}}: {{.}} [{{testArgs}}] - done
One arg: {{$}}: {{.}} [{{testArgs one}}] - done
Multiple arguments: {{$}}: {{.}} [{{testArgs aa bb cc 'x' " ! {[_.| }}] - done
Evil: {{$}}: {{.}} [{{testArgs bla; cat /etc/issue}}] - done
{{/arr2}}
{{/arr1}}
