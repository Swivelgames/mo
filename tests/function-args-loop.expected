No args: [test1 - ] - done
One arg: [test1 - one] - done
Multiple arguments: [test1 - aa bb cc 'x' " ! {[_.|] - done
Evil: [test1 - bla; cat /etc/issue] - done
No args: [test2 - ] - done
One arg: [test2 - one] - done
Multiple arguments: [test2 - aa bb cc 'x' " ! {[_.|] - done
Evil: [test2 - bla; cat /etc/issue] - done
